app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -865,11 +866,19 @@ class BrowserTabViewModel(
         val permissionEntity = locationPermissionsRepository.getDomainPermission(domain)
         permissionEntity?.let {
             if (it.permission == LocationPermissionType.ALLOW_ALWAYS) {
-                command.postValue(ShowDomainHasPermissionMessage(domain))
+                if (!locationPermissionMessages.containsKey(domain)) {
+                    setDomainHasLocationPermissionShown(domain)
+                    command.postValue(ShowDomainHasPermissionMessage(domain))
+                }
             }
         }
     }
 
+    @VisibleForTesting
```

Comment:
is there anything we can do to avoid this?

Inside the test, instead of using `setDomainHasLocationPermissionShown`, what will happen if we call twice `loadUrl("https://www.example.com", isBrowserShowing = true)`? I expect the first call to set `locationPermissionMessages[domain] = true`, and the second one not showing the Toast. Is that correct?

Reply:
Not really, `loadUrl` does a whole other bunch of stuff, triggering several `onPageChanged` and `onPageUpdated` that defies the purpose. That was my initial approach but it doesn't work.